### PR TITLE
TRUNK-5472 Add unit tests and refactor Allergy#hasSameValues

### PR DESCRIPTION
**Pull request by michalfotyga**
## Description of what I changed

Added unit tests for a `Allergy#hasSameValues(Allergy)` method.
When tests documented behavior I have refactored this method a little bit. Mostly split it into smaller methods so that now it reads better 💪. It's pure refactoring, no change in behavior.
These changes should fix [Sonar issue](https://sonar.openmrs.org/project/issues?id=org.openmrs%3Aopenmrs&issues=ae2156c1-38ea-4458-b1a0-996da5b0b35d&open=ae2156c1-38ea-4458-b1a0-996da5b0b35d)
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5472

## Checklist: I completed these to help reviewers :)

- [X] My pull request only contains **ONE single commit**

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

- [X] I have **added tests** to cover my changes. 

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [X] All new and existing **tests passed**.

- [X] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: api/src/main/java/org/openmrs/Allergy.java [hunk 1]
```diff
@@ -227,63 +227,72 @@ public Date getDateLastUpdated() {
...
+	}
+
 	/**
 	 * Checks if this allergy has the same reaction values as those in the given one
 	 * 
 	 * @param allergy the allergy who reaction values to compare with
 	 * @return true if the values match, else false
 	 */
 	private boolean hasSameReactions(Allergy allergy) {
-		if (getReactions().size() != allergy.getReactions().size()) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Allergy.java]: Any reason why you replaced getReactions()?

**Comment by dkayiwa** [comment]
@michalfotyga are you still working on this?

**Comment by samuelmale** [comment]
@michalfotyga  I'm closing this for now, but when you get more time feel free to reopen it.

**Comment by dkayiwa** [comment]
@samuelmale can you also make the ticket ready for work?

**Pull request closed without merging**